lib/seattleflu/id3c/cli/command/etl/redcap.py

```diff
@@ -0,0 +1,45 @@
+"""
+Functions shared by REDCap DET ETL
+"""
+from typing import Optional
+
+
+def normalize_net_id(net_id: str=None) -> Optional[str]:
+    """
+    If netid or UW email provided, return netid@washington.edu
+
+    >>> normalize_net_id('abcd')
+    'abcd@washington.edu'
+
+    >>> normalize_net_id('aBcD ')
+    'abcd@washington.edu'
+
+    >>> normalize_net_id('abcd@uw.edu')
+    'abcd@washington.edu'
+
+    >>> normalize_net_id('aBcD@u.washington.edu')
+    'abcd@washington.edu'
+
+    If missing netid or non-uw email provided, it cannot be normalized so return nothing
+    >>> normalize_net_id('notuw@gmail.com')
+    >>> normalize_net_id('nodomain@')
+    >>> normalize_net_id()
+    >>> normalize_net_id('')
+    >>> normalize_net_id(' ')
+    """
+
+    if not net_id or net_id.isspace():
+        return None
+
+    net_id = net_id.strip().lower()
+
+    # if a uw email was entered, drop the domain before normalizing
+    if '@' in net_id:
+        domain = net_id.split('@')[1]
```

Comment:
Could you split just once like net_id, domain =  net_id.split('@')

Reply:
That would only work if '@' is in the string. 
Otherwise, you'd get
```
a, b = 'aaaaa'.split('@')
ValueError: not enough values to unpack (expected 2, got 1)
```

Reply:
Line 37 ensures that:
if '@' in net_id:

Reply:
Ah I misunderstood. I thought you wanted to replace the conditional and the split with one line.

Reply:
@jccraft-uw I can parse the split list into `net_id, domain`, I'll just have to be more explicit about enforcing exactly one @ in the provided string and throw out edge case strings with multiple@at@signs

Reply:
By no means should you spend time trying to do it this way. It just seemed like in its current state we could split once instead of twice.

Reply:
@kschwabefry Instead of checking for multiple `@`, you can provide a limit to `str.split`: 

```python
local_part, domain = net_id.split('@', 1)
```